rules/CodingStyle/Application/UseImportsAdder.php

```diff
@@ -120,36 +129,45 @@ public function addImportsToNamespace(
 
         $existingUseImportTypes = $this->typeFactory->uniquateTypes($existingUseImportTypes);
 
-        $useImportTypes = $this->diffFullyQualifiedObjectTypes($useImportTypes, $existingUseImportTypes);
+        $newUseImportTypes = $this->diffFullyQualifiedObjectTypes($useImportTypes, $existingUseImportTypes);
+
+        $newFunctionUseImportTypes = $this->diffFullyQualifiedObjectTypes(
+            $functionUseImportTypes,
+            $existingFunctionUseImportTypes
+        );
 
-        $constantUseImportTypes = $this->diffFullyQualifiedObjectTypes(
+        $newConstantUseImportTypes = $this->diffFullyQualifiedObjectTypes(
             $constantUseImportTypes,
             $existingConstantUseImportTypes
         );
 
-        $functionUseImportTypes = $this->diffFullyQualifiedObjectTypes(
-            $functionUseImportTypes,
-            $existingFunctionUseImportTypes
+        $newUses = $this->createUses(
+            $newUseImportTypes,
+            $newFunctionUseImportTypes,
+            $newConstantUseImportTypes,
+            $namespaceName
         );
 
-        $newUses = $this->createUses($useImportTypes, $constantUseImportTypes, $functionUseImportTypes, $namespaceName);
-
         if ($newUses === []) {
             return;
         }
 
-        $this->mirrorUseComments($namespace->stmts, $newUses);
+        $useComments = $this->getAndRemoveExistingUseComments($namespace->stmts);
+
+        $namespace->stmts = $this->insertUseNodesInStatements($namespace->stmts, $newUses);
 
-        $namespace->stmts = array_merge($newUses, $this->resolveInsertNop($namespace), $namespace->stmts);
-        $namespace->stmts = array_values($namespace->stmts);
+        if ($useComments !== []) {
+            $this->reapplyUseComments($useComments, $namespace->stmts);
+        }
     }
 
     /**
+     * @param Stmt[] $stmts
      * @return Nop[]
      */
-    private function resolveInsertNop(FileWithoutNamespace|Namespace_ $namespace): array
+    private function resolveInsertNop(array $stmts, int $position): array
```

Comment:
We modify this function slightly to account for the fact that the Nop may have to be inserted at some position which is not 0